pkg/controller/node/nodecontroller.go

```diff
@@ -294,7 +294,20 @@ func NewNodeController(
 					return
 				}
 
-				node := obj.(*api.Node)
+				node, isNode := obj.(*api.Node)
+				// We can get DeletedFinalStateUnknown instead of *api.Node here and we need to handle that correctly. #34692
+				if !isNode {
+					deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+					if !ok {
+						glog.Warningf("Received unexpected object: %v", obj)
```

Comment:
Errorf (if we don't get neither node nor deletedfinalstateunknown, then we really have problem)